### ISSUE TITLE
[TRAINER] GridSearch / RandomSearch of the hyperparameters method

### ISSUE THREAD

**Issue by art-test-stack** [art-test-stack/vessel_position_forecast#4]
Implement GridSearch / RandomSearch of the hyperparameters method
- the search is based on (X_train, y_train)
- evaluation is based on (X_test, y_test) using self.metric (Trainer.eval())

-> Difficulty : have to adapt model class to fit scikit-learn requirements for model learning

return best hyperparameters
